diff --git a/src/ftxui/screen/terminal.cpp b/src/ftxui/screen/terminal.cpp
index feeeba95..6d53f165 100644
--- a/src/ftxui/screen/terminal.cpp
+++ b/src/ftxui/screen/terminal.cpp
@@ -14,7 +14,8 @@
 #endif
 
 #include <windows.h>
-#else
+#endif
+#ifndef __SWITCH__
 #include <sys/ioctl.h>  // for winsize, ioctl, TIOCGWINSZ
 #include <unistd.h>     // for STDOUT_FILENO
 #endif
@@ -107,6 +108,8 @@ Dimensions Size() {
   }
 
   return FallbackSize();
+#elif defined(__SWITCH__)
+  return Dimensions{80, 45};
 #else
   winsize w{};
   const int status = ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);  // NOLINT
