diff --git a/src/ftxui/component/screen_interactive.cpp b/src/ftxui/component/screen_interactive.cpp
index 6dc982f0..f22288ea 100644
--- a/src/ftxui/component/screen_interactive.cpp
+++ b/src/ftxui/component/screen_interactive.cpp
@@ -47,7 +47,9 @@
 #else
 #include <fcntl.h>
 #include <sys/select.h>  // for select, FD_ISSET, FD_SET, FD_ZERO, fd_set, timeval
+#ifndef __SWITCH__
 #include <termios.h>  // for tcsetattr, termios, tcgetattr, TCSANOW, cc_t, ECHO, ICANON, VMIN, VTIME
+#endif
 #include <unistd.h>  // for STDIN_FILENO, read
 #include <cerrno>
 #endif
@@ -592,7 +594,8 @@ void ScreenInteractive::Install() {
 
   SetConsoleMode(stdin_handle, in_mode);
   SetConsoleMode(stdout_handle, out_mode);
-#else  // POSIX (Linux & Mac)
+#elif defined(__SWITCH__)  
+#else  /// POSIX (Linux & Mac)
   // #if defined(__EMSCRIPTEN__)
   //// Reading stdin isn't blocking.
   // int flags = fcntl(0, F_GETFL, 0);
diff --git a/src/ftxui/screen/terminal.cpp b/src/ftxui/screen/terminal.cpp
index feeeba95..6d53f165 100644
--- a/src/ftxui/screen/terminal.cpp
+++ b/src/ftxui/screen/terminal.cpp
@@ -14,7 +14,8 @@
 #endif
 
 #include <windows.h>
-#else
+#endif
+#ifndef __SWITCH__
 #include <sys/ioctl.h>  // for winsize, ioctl, TIOCGWINSZ
 #include <unistd.h>     // for STDOUT_FILENO
 #endif
@@ -107,6 +108,8 @@ Dimensions Size() {
   }
 
   return FallbackSize();
+#elif defined(__SWITCH__)
+  return Dimensions{80, 45};
 #else
   winsize w{};
   const int status = ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);  // NOLINT
